### PR TITLE
feat:[CDS-72444]: optimization in encryption of tf plan

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Changes are regarding optimisation in encryption of tf plan for harness sm where we are uploading file on manager side.
Tests Done: 
https://docs.google.com/spreadsheets/d/1AlleguoCSfOBfAC7aDsSgR77q4YkRhG0aUCfsiONuVs/edit#gid=0

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49717)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/beans/DelegateFileEncryptedDataPackage.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+public class DelegateFileEncryptedDataPackage {
+  EncryptData encryptData;
+  DelegateFile delegateFile;
```
Comment (yogesh-chauhan) [inline 930-delegate-tasks/src/main/java/io/harness/beans/DelegateFileEncryptedDataPackage.java:22]: can we create a new class or directly use the needed fields from DelegateFile class ?

**Comment by vlica-harness** [comment]
@sourabh-awashti can you please check this TerragruntTaskService class,
method: saveTerraformPlanContentToFile
Are we not missing the proper decryption for NG there?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/handlers/HarnessSMEncryptionDecryptionHandlerNG.java [hunk 2]
```diff
@@ -69,33 +68,38 @@ record = execute(delegateManagerEncryptionDecryptionHarnessSMClient.encryptHarne
...
     }
     long endTime = Instant.now().getEpochSecond();
     log.info(format("Encryption of TF plan completed with total time taken in seconds : %s", endTime - startTime));
-    return decryptedRecord;
+
+    record.setUuid(generateUuid());
+    return record;
+  }
+
+  public byte[] getDecryptedContent(EncryptionConfig config, EncryptedRecordData record) throws IOException {
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/handlers/HarnessSMEncryptionDecryptionHandlerNG.java:100]: @sourabh-awashti can you please check this TerragruntTaskService class,
method: saveTerraformPlanContentToFile
Are we not missing the proper decryption for NG there?

Comment (sourabh-awashti) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/handlers/HarnessSMEncryptionDecryptionHandlerNG.java:100]: I have handled the encryption on manager in this file for this method

Comment (vlica-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/handlers/HarnessSMEncryptionDecryptionHandlerNG.java:100]: is the isNG method parameter needed then in saveTerraformPlanContentToFile?

Comment (sourabh-awashti) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/terraform/handlers/HarnessSMEncryptionDecryptionHandlerNG.java:100]: yes, it is not needed as this is getting called from NG only

File: 400-rest/src/main/java/software/wings/beans/DelegateFileEncryptedDataPackage.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+package software.wings.beans;
+
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.beans.EncryptData;
+import io.harness.delegate.beans.DelegateFile;
+
+import lombok.Builder;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
```
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 400-rest/src/main/java/software/wings/beans/DelegateFileEncryptedDataPackage.java]: done

File: 400-rest/src/main/java/software/wings/resources/DelegateManagerEncryptionDecryptionHarnessSMResource.java [hunk 4]
```diff
@@ -65,4 +68,20 @@ public RestResponse<DecryptedRecord> decryptHarnessSMSecret(
...
+  @Path("/encrypt-harness-sm-secret-upload")
+  public RestResponse<DelegateFileEncryptedRecordDataPackage> encryptHarnessSMSecretWithFileUpload(
+      @NotNull @QueryParam("accountId") String accountId,
+      @NotNull DelegateFileEncryptedDataPackage delegateFileEncryptedDataPackage) throws IOException {
+    try {
+      return new RestResponse<>(encryptionDecryptionService.encryptDataWithFileUpload(accountId,
+          delegateFileEncryptedDataPackage.getEncryptData().getContent(),
+          delegateFileEncryptedDataPackage.getDelegateFile()));
+    } catch (Exception e) {
+      log.error(format("Unable to encrypt the content for harness secret manager for account %s", accountId), e);
```
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 400-rest/src/main/java/software/wings/resources/DelegateManagerEncryptionDecryptionHarnessSMResource.java]: done

File: 400-rest/src/main/java/software/wings/resources/DelegateManagerEncryptionDecryptionHarnessSMResourceNG.java [hunk 5]
```diff
@@ -52,6 +55,22 @@ public RestResponse<EncryptedRecordData> encryptHarnessSMSecretNG(
...
+  public RestResponse<DelegateFileEncryptedRecordDataPackage> encryptHarnessSMSecretNGWithFileUpload(
+      @NotNull @QueryParam("accountId") String accountId,
+      @NotNull DelegateFileEncryptedDataPackage delegateFileEncryptedDataPackage) throws IOException {
+    try {
+      return new RestResponse<>(encryptionDecryptionService.encryptDataNGWithFileUpload(accountId,
+          delegateFileEncryptedDataPackage.getEncryptData().getContent(),
+          delegateFileEncryptedDataPackage.getDelegateFile()));
+    } catch (Exception e) {
+      log.error(format("Unable to encrypt the content for harness secret manager for account %s", accountId), e);
+      throw e;
```
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 400-rest/src/main/java/software/wings/resources/DelegateManagerEncryptionDecryptionHarnessSMResourceNG.java:72]: done

File: 930-delegate-tasks/src/main/java/io/harness/beans/DelegateFileEncryptedDataPackage.java (hunk 1 above)
Comment (sourabh-awashti) [reply 930-delegate-tasks/src/main/java/io/harness/beans/DelegateFileEncryptedDataPackage.java]: done

**Review by yogesh-chauhan — APPROVED** [review]

**Review by vlica-harness — APPROVED** [review]

**Review by vlica-harness — APPROVED** [review]

**Comment by sourabh-awashti** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 13 files at r1, 13 of 17 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49717)* status: 18 file(s) to review. 1. unresolved discussions.

**Comment by sourabh-awashti** [comment]
trigger review

**Pull request merged**